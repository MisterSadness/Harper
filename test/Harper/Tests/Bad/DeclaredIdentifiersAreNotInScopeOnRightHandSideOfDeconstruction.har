main :: sideeffect -> ();
main = {
  // If lambdas could be recursive, i.e. `f` was visible in the body of the deconstructed lambda,
  // this code would have no sensible semantics, as `f` would be used during the evaluation 
  // of the value being assigned to it. This restriction could theoretically be dropped if the rhs is pure.
  (f :: Integer -> Integer) = (\(x :: Integer) => {
      eval printLn (f 10) ();
      return (\(n :: Integer) => n * x);
      }) 42 ();

  eval printLn (f 10) ();
};