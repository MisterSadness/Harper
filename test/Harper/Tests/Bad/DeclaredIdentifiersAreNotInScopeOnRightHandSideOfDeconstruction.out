Error: undeclared identifier `f`.
During evaluation of:
 f
Located at line 7 column 21

Execution terminated with an error: type error.